probe-rs/src/debug/unit_info.rs

```diff
@@ -2213,6 +2203,45 @@ impl UnitInfo {
             offset: offset.unwrap_or(BitOffset::FromLsb(0)),
         }))
     }
+
+    fn extract_source_location(
+        &self,
+        debug_info: &DebugInfo,
+        entry: &gimli::DebuggingInformationEntry<GimliReader>,
+    ) -> Result<Option<SourceLocation>, gimli::Error> {
+        let Some(file_attr) = entry.attr_value(gimli::DW_AT_decl_file)? else {
+            return Ok(None);
+        };
+
+        let Some(path) = extract_file(debug_info, &self.unit, file_attr) else {
+            return Ok(None);
+        };
+
+        let mut source_location = SourceLocation {
+            path,
+            line: None,
+            column: None,
+        };
+
+        let mut variable_attributes = entry.attrs();
+        // Now loop through all the unit attributes to extract the remainder of the `Variable` definition.
+        while let Ok(Some(attr)) = variable_attributes.next() {
+            match attr.name() {
+                gimli::DW_AT_decl_line => {
+                    if let Some(line_number) = extract_line(attr.value()) {
+                        source_location.line = Some(line_number);
+                    }
+                }
+                gimli::DW_AT_decl_column => {
+                    // Unused.
```

Comment:
I know you didn't introduce this, but why is this unused?

Reply:
It seems to be missing from the debug info generated by rustc, but I've added support for it now. At leat the C code might benefit.